### PR TITLE
Tsql not equal workaround

### LINKED ISSUE #3001: condA != condB not supported in TSQL
`snapshot_check_strategy()` is not dispatched, which means it can only be overridden in a custom adapter. Which was working for `dbt-sqlserver` but we can't import that overwritten macro into `dbt-synapse` like we're doing with all the other properly dispatched macros.

rather than dispatching the entire `snapshot_check_strategy()`, we can, with a simple tweak to logic, modify it so that it works on TSQL but doesn't break the existing adapters.

The benefit is a lower code footprint which means the dbt-msft adapters can stay more current with `dbt-core`

### PR DESCRIPTION
**Pull request by swanderz**
resolves #3001 
@jtcohen6 @mikaelene 

### Description

per this conversation
https://github.com/dbt-msft/dbt-synapse/pull/32#discussion_r553758414

### Checklist
 - [x] I have signed the [CLA](https://docs.getdbt.com/docs/contributor-license-agreements)
 - [x] I have run this code in development and it appears to resolve the stated issue
 - [x] This PR includes tests, or tests are not required/relevant for this PR
 - [x] I have updated the `CHANGELOG.md` and added information about my change to the "dbt next" section.

### CODE REVIEW

**Comment by swanderz** [comment]
strange redshift error.. perhaps it is transient?
```
=================================== FAILURES ===================================
_ IncrementalTestConcurrentTransaction.test__redshift__concurrent_transaction_incremental _
[gw3] linux -- Python 3.6.9 /home/dbt_test_user/project/.tox/integration-redshift-py36/bin/python

self = <test_concurrent_transaction.IncrementalTestConcurrentTransaction testMethod=test__redshift__concurrent_transaction_incremental>

    @use_profile("redshift")
    def test__redshift__concurrent_transaction_incremental(self):
        self.reset()
>       self.run_test()

test/integration/032_concurrent_transaction_test/test_concurrent_transaction.py:136: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
test/integration/032_concurrent_transaction_test/test_concurrent_transaction.py:102: in run_test
    self.assertEqual(self.query_state['view_model'], 'good')
E   AssertionError: 'error: UDF working memory threshold excee[280 chars]\n\n' != 'good'
E   + good- error: UDF working memory threshold exceeded
E   - DETAIL:  
E   -   -----------------------------------------------
E   -   error:  UDF working memory threshold exceeded
E   -   code:      10002
E   -   context:   
E   -   query:     0
E   -   location:  stream.cpp:41
E   -   process:   padbmaster [pid=2255]
E   -   -----------------------------------------------
E   -
```

**Comment by swanderz** [comment]
also git newb... but I made my two commits into a new branch directly after running `git reset --hard upstream/dev/kiyoshi-kuromiya`, so i'm confused why those 6 commits from two years ago are showing up...

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,7 +1,11 @@
 ## dbt 0.19.0 (Release TBD)
 
+### Under the hood
+- rewrite logic in `snapshot_check_strategy()` to make compatible with TQL [#3000](https://github.com/fishtown-analytics/dbt/pull/3000)
```
Comment (jtcohen6) [inline CHANGELOG.md:4]:
```suggestion
- Rewrite logic in `snapshot_check_strategy()` to make compatible with other SQL dialects ([#3000](https://github.com/fishtown-analytics/dbt/pull/3000), [#3001](https://github.com/fishtown-analytics/dbt/pull/3001))
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,7 +1,11 @@
...
+
 Contributors:
 - [@dbeatty10](https://github.com/dbeatty10) ([dbt-adapter-tests#10](https://github.com/fishtown-analytics/dbt-adapter-tests/pull/10))
+- [@swanderz](https://github.com/swanderz) [#3000](https://github.com/fishtown-analytics/dbt/pull/3000)
```
Comment (jtcohen6) [inline CHANGELOG.md:8]:
```suggestion
- [@swanderz](https://github.com/swanderz) ([#3000](https://github.com/fishtown-analytics/dbt/pull/3000))
```

File: core/dbt/include/global_project/macros/materializations/snapshot/strategies.sql [hunk 3]
```diff
@@ -164,7 +164,14 @@
     {%- for col in check_cols -%}
         {{ snapshotted_rel }}.{{ col }} != {{ current_rel }}.{{ col }}
         or
-        ({{ snapshotted_rel }}.{{ col }} is null) != ({{ current_rel }}.{{ col }} is null)
+        {# hack to make CI try again #}
+        {# below is equivalent to the two clauses being not equal (!=) to one another  #}
+        {# ({{ snapshotted_rel }}.{{ col }} is null) != ({{ current_rel }}.{{ col }} is null) #}
```
Comment (jtcohen6) [inline core/dbt/include/global_project/macros/materializations/snapshot/strategies.sql:169]: Remove comments
```suggestion
```

**Review by jtcohen6 — COMMENTED** [review]
Nice! Ok, I'm up for sneaking this into v0.19.0, since we already have some eyes on snapshots (on account of `invalidate_hard_deletes`).

Copying from https://github.com/dbt-msft/dbt-synapse/pull/32:
```sql
select

  scenario,
  (a is null) != (b is null) as previous_logic,
  ((a is null) and not (b is null)) or (not (a is null) and (b is null)) as proposed_logic

from (

  select 1 as scenario, null as a, null as b
  union all
  select 2 as scenario, 1 as a, 1 as b
  union all
  select 3 as scenario, null as a, 1 as b
  union all
  select 4 as scenario, 1 as a, null as b

) sbq
```


scenario | previous_logic | proposed_logic
-- | -- | --
1 | false | false
2 | false | false
3 | true | true
4 | true | true

This works on every database I can think to test: Postgres, Redshift, Snowflake, BigQuery, Spark, Presto.

Looks like the Redshift error was intermittent—sometimes the UDF memory is exhausted, we're not exactly sure why—so I reran the tests with success.

Tiny tiny requested changes re: changelog note + comments. As far as the extraneous git commit, I think those are hanging out on your fork's version of `dev/kiyoshi-kuromiya` ([see diff](https://github.com/fishtown-analytics/dbt/compare/dev/kiyoshi-kuromiya...swanderz:dev/kiyoshi-kuromiya)). You could try:
- Deleting your fork's branch and recreating from remote
- Rebasing your branch against `fishtown-analytics/dbt/dev/kiyoshi-kuromiya`

**Review by jtcohen6 — APPROVED** [review]

**Comment by mikaelene** [comment]
This is great! Will make dbt-sqlservers life much easier

**Pull request merged**